webapp/src/main/java/org/apache/atlas/web/rest/TypesREST.java

```diff
@@ -540,15 +566,22 @@ public void deleteAtlasTypeDefs(final AtlasTypesDef typesDef) throws AtlasBaseEx
     @Timed
     public void deleteAtlasTypeByName(@PathParam("typeName") final String typeName) throws AtlasBaseException {
         AtlasPerfTracer perf = null;
-        typeCacheRefresher.verifyCacheRefresherHealth();
         InterProcessMutex lock = null;
+        RequestContext.get().setTraceId(UUID.randomUUID().toString());
         try {
+            typeCacheRefresher.verifyCacheRefresherHealth();
             if (AtlasPerfTracer.isPerfTraceEnabled(PERF_LOG)) {
                 perf = AtlasPerfTracer.getPerfTracer(PERF_LOG, "TypesREST.deleteAtlasTypeByName(" + typeName + ")");
             }
             lock = attemptAcquiringLock();
             typeDefStore.deleteTypeByName(typeName);
             typeCacheRefresher.refreshAllHostCache();
+        } catch (AtlasBaseException atlasBaseException) {
+            LOG.error("TypesREST.deleteAtlasTypeByName:: " + atlasBaseException.getMessage(), atlasBaseException);
+            throw atlasBaseException;
+        } catch (Exception e) {
+            LOG.error("TypesREST.deleteAtlasTypeByName:: " + e.getMessage(), e);
```

Comment:
This catch block will also catch AtlasBaseException right? So why do we need a seperate catch block above for AtlasBaseException.
Throwing the exception will log it in Sentry and application logs right, why are we logging the error separately here

Reply:
If a method throws AtlasBaseExcecption then we do want to re-throw the same so that HTTP response can have error code and message. If we keep only single catch block with Exception type then it will swallow actual type of the exception and result in internal_server_error in the HTTP response.